mlflow/_spark_autologging.py

```diff
@@ -217,6 +220,8 @@ def _notify(self, path, version, data_format):
         Method called by Scala SparkListener to propagate datasource read events to the current
         Python process
         """
+        if autologging_is_disabled("spark"):
```

Comment:
@dbczumar  I need access to the flavor name but the problem is if I define `FLAVOR_NAME` variable then https://github.com/mlflow/mlflow/blob/master/tests/test_flavors.py#L30 is picking up `mlflow/_spark_autologging.py` as a flovor as opposed to just `mlflow/spark.py` and fails the test. I am still trying to think of a better way to pass in flavor name to `_spark_autologging.py`

Reply:
@mohamad-arabi Nice catch! Can we update https://github.com/mlflow/mlflow/blob/master/tests/test_flavors.py#L30 to only pick up public modules (e.g., modules that aren't prefixed with `_`?). That should solve our problem of importing `FLAVOR_NAME` in `_spark_autologging`.

Reply:
Not really @dbczumar , as you can see in line 31, it expects one of the files to be `__init__.py`.  will check with Haru, if we can filter out files starting with `_` and remove the if case for `__init__.py`. Not sure why it is there

https://github.com/mlflow/mlflow/blob/master/tests/test_flavors.py#L31

Reply:
As discussed offline, we'll keep `__init__` as a special case and otherwise omit any private modules beginning with the `_` character.